tests/integration/test_case.go

```diff
@@ -349,8 +346,9 @@ type UpdateDoc struct {
 	// contains this string.
 	ExpectedError string
 
-	// Setting DontSync to true will prevent waiting for that update.
-	DontSync bool
+	// Skip waiting for an update event. This should only be used for
+	// tests that do not correctly publish an update event.
+	SkipUpdateEvent bool
```

Comment:
suggestion: why is it synced by default now? That resembles sequentially consistent memory order instead of relaxed one that we used before.
I think, if we want to sync, we should request it explicitly. Unsynced actions are closer to realistic behaviour.

Reply:
todo: I think this is a misleading name and comment, IIRC `waitForUpdateEvents` does not wait for the P2P sync, but only the update event to be published to the event bus.  If that is correct, @nasdf can you please rework the property name and documentation to make that clearer.

Reply:
what is this then if not wait: `<-s.nodeEvents[i].update.Message():`?

Reply:
That is just the event bus, it is written to synchronously by the database when mutating documents - the P2P code also subscribes to this bus - so 'waiting' to the bus should never really block, and the P2P stuff hasn't even been sent over the network at that point.

If I understand it correctly.  I made the same comment as you did earlier for `waitForUpdateEvents` - I think Keenan improved the documentation and code location, but given that it is still tripping up reviewers, that too should probably be improved further.